### DISCUSSION TITLE
Hot Reloading

### DISCUSSION THREAD

**Discussion by knoopx** [idom-team/idom#837] [Question] [answered]
Does idom support any kind of hot reloading? can it be used with other python reloading mechanisms?

**Comment by rmorshea** [comment, marked as answer]
As explained [here](https://idom-docs.herokuapp.com/docs/guides/getting-started/running-idom.html#running-idom), IDOM can be run using a [variety of different web servers](https://idom-docs.herokuapp.com/docs/guides/getting-started/installing-idom.html#native-backends). As such, whether hot-reloading is supported and how to enable it depend on you choice of server. For example, to enable hot-reloading in Flask, we can write the following:

```python
# hello.py

from flask import Flask

from idom import component, html
from idom.backend.flask import configure

@component
def HelloWorld():
    return html.h1("Hello, world!")

app = Flask(__name__)
configure(app, HelloWorld)
```

And then follow the [flask documentation](https://flask.palletsprojects.com/en/2.2.x/cli/#debug-mode) on how to run with hot-reloading:

```bash
flask --app hello --debug run
```

If you chose a different server implementation, this answer would change. Like if you used one of the supported ASGI servers (e.g. [FastAPI](https://idom-docs.herokuapp.com/docs/guides/getting-started/running-idom.html#running-with-fastapi)) and ran it with [uvicorn](https://www.uvicorn.org/), you'd need to follow Uvicorn's directions on [reloading behavior](https://www.uvicorn.org/settings/#development) instead.

**Reply by allanleal** [reply]
It would be nice if the `run` function had an argument `autoreload` or `hotreload` that could be set `True`, instead of using the `configure` approach above.

**Reply by rmorshea** [reply]
That would definitely be preferable. I think that we may even turn that on by default since the `run` function is explicitly meant to be used during development and not in production. https://github.com/reactive-python/reactpy/issues/990
